src/runner.js

```diff
@@ -103,50 +102,67 @@ const runScenario = (stepsInstance, scenario, testMode) => {
+}
 
-    const itemTags = item.tags.map(tag => tag.replace('@', ''))
-    return tags.find(tag => itemTags.includes(tag.replace('@', '')))
-  }, [])
+/**
+ * @param {string?} name - name of item (feature|scenario) to check
+ * @param {string | Array<string>} tags - tags of item (feature|scenario) to check
+ * @param {string?} filterOptions.name - name filter
+ * @param {string | Array<string>} filterOptions.tags - tags filter
+ * @return {Boolean} - true if feature matches the filter options
+ */
+const itemMatch = (name='', tags=[], filterOptions={}) => {
+  const { name: filterName, tags: filterTags } = filterOptions
+
+  const parsedTags = isStr(filterTags) 
+    ? parseFeatureTags(filterTags) 
+    : eitherArr(filterTags, [])
+
+  const nameMatch = !filterName || name.includes(filterName)
+  const tagMatch = !parsedTags.length || parsedTags.every(clientTag => tags.includes(clientTag))
+
+  return nameMatch && tagMatch
 }
 
 /**
- * Filters features and scenarios based on the passed in tag options
+ * Filters features and scenarios based on the passed in filterOptions
  * @function
  * @private
  * @param {Array} features - Features to be run
- * @param {Array} tags - Tags to filter which Features and scenarios will be run
+ * @param {Object} tags - Tags to filter which Features and scenarios will be run
+ *  * @param {string?} filterOptions.name - name of feature 
+ * @param {string | Array<string>} filterOptions.tags - feature tags to match
  *
  * @returns {Array} - Filtered features that should be run
  */
-const filterFromTags = (features, tags) => {
-  // If no tags, then run all features
-  return !tags || (isArr(tags) && !tags.length)
-    ? features
-    : // Get all the tagged features, removing any non-tagged features
-    getTaggedItems(features, tags).reduce((filtered, feature) => {
-      // Get all tagged scenarios, removing any non-tagged scenarios
-      const tagScenarios = getTaggedItems(feature.scenarios, tags)
-
-      // If tagged scenarios were
-      //  * FOUND - Only include the matching tagged scenarios
-      //  * NOT FOUND - All scenarios will be run, because the feature had a matching tag
-      tagScenarios.length
-        ? filtered.push({ ...features, scenarios: tagScenarios })
-        : filtered.push(feature)
-
+const filterFeatures = (features, filterOptions={}) => {
+  return features.reduce((filtered, feature) => {
+    const isMatchingFeature = itemMatch(feature.feature, feature.tags, filterOptions)
+    if (isMatchingFeature) {
+      filtered.push(feature)
       return filtered
-    }, [])
+    }
+
+    // check for matching scenarios, where scenarios inherit their parent feature's tags
+    const matchingScenarios = feature.scenarios.filter(
+      scenario => itemMatch(
+        scenario.scenario, 
+        [ ...(scenario.tags || []), ...(feature.tags || []) ],
+        filterOptions
+      )
+    )
+    if (matchingScenarios.length) {
+      feature.scenarios = matchingScenarios
```

Comment:
This should create a new feature object instead of mutating the current one.
Since you can pass a feature object in, we don't own it, so we should not mutate it
Just do this instead => `filtered.push({ ... feature, scenarios: matchingScenarios })` on line 162

Reply:
lol I had it that way at first, but was so certain you would argue it would be better to mutate for performance reasons, so I changed it to a mutation.

Updated 👍🏼